ivy_tests/test_ivy/helpers/hypothesis_helpers/general_helpers.py

```diff
@@ -234,6 +235,43 @@ def get_shape(
     return tuple(shape)
 
 
+def tensorflow_gradients_args(*,
+                              input_shapes=None,
+                              input_dtypes=None,
+                              output_grad_shapes=None,
+                              output_grad_dtypes=None):
+
+        """
+    Generates arguments for testing the gradients function of a TensorFlow frontend.
+    Parameters
+    ----------
+    input_shapes : tuple of tuples of ints, optional
+        Shapes of the input tensors. If not specified, defaults to ((1,),).
+    input_dtypes : tuple of dtypes, optional
+        Data types of the input tensors. If not specified, defaults to (tf.float32,).
+    output_grad_shapes : tuple of tuples of ints, optional
+        Shapes of the output gradient tensors. If not specified, defaults to None.
+    output_grad_dtypes : tuple of dtypes, optional
+        Data types of the output gradient tensors. If not specified, defaults to None.
+    Returns
+    -------
+    A tuple of the form (inputs, output_grads), where:
+    - inputs is a tuple of input tensors.
+    - output_grads is a tuple of output gradient tensors of the same shape as the output tensors.
+    """
+        input_shapes = input_shapes or ((1,),)
+        input_dtypes = input_dtypes or (tf.float32,)
+        output_grad_shapes = output_grad_shapes or input_shapes
+        output_grad_dtypes = output_grad_dtypes or input_dtypes
+
+        inputs = tuple(tf.ones(shape=shape, dtype=dtype) for shape, dtype in zip(input_shapes, input_dtypes))
+        output_tensors = tf.gradients(tf.reduce_sum(tf.stack([tf.reduce_prod(inp) for inp in inputs])), inputs)
+
+        output_grads = tuple(tf.ones(shape=shape, dtype=dtype) for shape, dtype in zip(output_grad_shapes, output_grad_dtypes))
+
+        return inputs, output_grads
```

Comment:
We shouldn't have framework or frontend specific strategies, instead we should use Hypothesis and Ivy strategies to generate valid arguments.

Reply:
ok sir I got that but,  think there are some issues with general_functions testing, I'm trying to run tests locally it always gives the `ERROR: not found`

Reply:
![image](https://github.com/unifyai/ivy/assets/54188431/06435420-1404-4678-9006-049186cdd0be)

Reply:
Hi @KartikeyDubeyKD . I think the issue is most likely related to an incorrect path in your pytest configuration for the test you're running. It's best to follow the guide regarding setting up testing as mentioned in the Docs [here](https://unify.ai/docs/ivy/overview/contributing/setting_up.html#setting-up-testing-in-pycharm)

Reply:
No @YushaArif99 other functions are running fine, I'm just facing a problem in this one only, can you please check once